public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -149,41 +149,44 @@ function legacyCreateSpanLinkFactory(
     }
 
     // Get metrics links
-    if (metricsDataSourceSettings && traceToMetricsOptions) {
+    if (metricsDataSourceSettings && traceToMetricsOptions?.queries) {
       const defaultQuery = `histogram_quantile(0.5, sum(rate(tempo_spanmetrics_latency_bucket{operation="${span.operationName}"}[5m])) by (le))`;
-      const dataLink: DataLink<PromQuery> = {
-        title: metricsDataSourceSettings.name,
-        url: '',
-        internal: {
-          datasourceUid: metricsDataSourceSettings.uid,
-          datasourceName: metricsDataSourceSettings.name,
-          query: {
-            expr: traceToMetricsOptions.query ?? defaultQuery,
-            refId: '',
+
+      links.metricLinks = [];
+      for (const query of traceToMetricsOptions.queries) {
+        const dataLink: DataLink<PromQuery> = {
+          title: metricsDataSourceSettings.name,
+          url: '',
+          internal: {
+            datasourceUid: metricsDataSourceSettings.uid,
+            datasourceName: metricsDataSourceSettings.name,
+            query: {
+              expr: query.query || defaultQuery,
+              refId: '',
```

Comment:
I realise you didn't make the change in this PR, but I wonder why we're adding an empty string for refId.

Reply:
It's a required field in the type, but I'm not sure what it does in explore. Updating it to 'A' which seems like a common default.